train.rb

```diff
@@ -0,0 +1,128 @@
+class Station
+  attr_reader :name, :trains
+
+  def initialize(name)
+    @name = name
+    @trains = []
+  end
+
+  def add_trains(train)
+    @trains << train
+  end
+
+  def dep_trains(train)
+    @trains.delete(train)
+  end
+
+  def train_type(type)
+    trains.count { |train| train.type == type }
+  end
+  
+end
+
+
+class Route
+  attr_reader :stations
+
+  def initialize(first, last)
+    @stations = [first, last]
+  end
+
+  def add_station(station) # добавление промежуточной станции
+    stations.insert(-2, station)
+  end
+
+  def remove_station(station) # удаление промежуточной станции
+    unless [@stations[0], @stations[-1]].include?(station)
```

Comment:
еще правильней было бы написать отдельные методы для этих станций
тогда здесь можно было бы так:
```suggestion
    unless [first_station, last_station].include?(station)
```